src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -21,520 +21,550 @@
+         }
+         return AIState.PREPARING;
+     }
+
+     /**
+      * Searches and adds a field that has not been taken yet for the farmer and then adds it to the list.
+      */
+     private void searchAndAddFields()
+     {
+         Colony colony = worker.getColony();
+         if (colony != null)
+         {
+             @Nullable Field newField = colony.getFreeField(worker.getName());
+
+             if (newField != null && getOwnBuilding() != null)
+             {
+                 getOwnBuilding().addFarmerFields(newField);
+             }
+         }
+     }
+
+     /**
+      * Checks if the farmer is ready to plant.
+      *
+      * @param currentField the field to plant.
+      * @return true if he is ready.
+      */
+     private boolean canGoPlanting(@NotNull Field currentField, @NotNull BuildingFarmer buildingFarmer)
+     {
+         if (currentField.getSeed() == null)
+         {
+             chatSpamFilter.talkWithoutSpam("entity.farmer.noSeedSet");
+             buildingFarmer.setCurrentField(null);
+             return false;
+         }
+
+         if (shouldTryToGetSeed)
+         {
+             if (walkToBuilding())
+             {
+                 return false;
+             }
+             isInHut(new ItemStack(currentField.getSeed()));
+             shouldTryToGetSeed = false;
+         }
+
+         return true;
+     }
+
+     /**
+      * Executes the hoeing of the field.
+      *
+      * @return the next state.
+      */
+     @NotNull
+     private AIState hoe()
+     {
+         @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+
+         if (buildingFarmer == null || buildingFarmer.getCurrentField() == null)
+         {
+             return AIState.PREPARING;
+         }
+         @Nullable Field field = getOwnBuilding().getCurrentField();
+
+         if (workingOffset != null)
+         {
+             BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
+             if (walkToBlock(position.up()))
+             {
+                 return AIState.FARMER_HOE;
+             }
+
+             if (shouldHoe(position, field))
+             {
+                 if (checkForHoe())
+                 {
+                     return AIState.PREPARING;
+                 }
+                 equipHoe();
+                 worker.swingArm(worker.getActiveHand());
+                 world.setBlockState(position, Blocks.FARMLAND.getDefaultState());
+                 worker.damageItemInHand(1);
+                 mineBlock(position.up());
+             }
+         }
+
+         if (!handleOffset(field))
+         {
+             resetVariables();
+             buildingFarmer.getCurrentField().setNeedsWork(true);
+             buildingFarmer.getCurrentField().setFieldStage(HOED);
+             return AIState.IDLE;
+         }
+         BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'position' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)